lib/bolt/pal.rb

```diff
@@ -163,9 +163,10 @@ def detect_project_conflict(project, environment)
     # Runs a block in a PAL script compiler configured for Bolt.  Catches
     # exceptions thrown by the block and re-raises them ensuring they are
     # Bolt::Errors since the script compiler block will squash all exceptions.
-    def in_bolt_compiler
+    def in_bolt_compiler(compiler_params: {})
       # TODO: If we always call this inside a bolt_executor we can remove this here
       setup
+      compiler_params.merge!(set_local_facts: false)
```

Comment:
```suggestion
      compiler_params = compiler_params.merge(set_local_facts: false)
```

I think we generally try to avoid a destructive merge for method parameters, since it'll modify the hash from the calling method as well.